sycl/doc/extensions/OnlineCompilation/OnlineCompilation.asciidoc

```diff
@@ -0,0 +1,200 @@
+= SYCL Intel extension: Online Compilation
+Konstantin Bobrovskii <konstantin.s.bobrovsky@intel.com>, John Pennycook <john.pennycook@intel.com>
+v0.1
+:source-highlighter: pygments
+:icons: font
+== Introduction
+This document describes an interface for online compilation from high-level languages, such as
+OpenCL, to a binary format, such as SPIRV, loadable by SYCL backends. Unlike SYCL'2020 OpenCL Back
+End's online comilation interface, this interface is not bound to any particular Back End and does
+not require available SYCL context for online compilation.
+
+This gives a flexibility to "cross-compile" to SPIRV or other supported formats without any SYCL
+device or context available. The online compilation API uses the `online_compiler` class to access
+compilation services. Instances of the class are constructed based on a specification of the desired
+compilation target passed to the constructors - such as compiled code format, target architecture,
+etc. All the settings are optional, and by default the target is generic SPIRV.
+
+This API is an Intel SYCL extension.
+
+== Online compilation API
+
+All online compilation API elements reside in the `sycl::INTEL` namepace.
+
+=== Source language specification
+
+Elements of the enum designate the source language:
+[source,c++]
+-----------------
+enum class source_language {
+  OpenCL_C, // OpenCL C language
+  CM        // Intel's C-for-Media language
+};
+-----------------
+
+=== APIs to express compilation target characteristics
+
+The desired format of the compiled code: 
+[source,c++]
+-----------------
+enum class compiled_code_format {
+  SPIRV
```

Comment:
Similar to above, maybe `compiled_code_format::spir_v`?

Reply:
ok